oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureBlobStoreBackend.java

```diff
@@ -0,0 +1,784 @@
+                        this,
+                        connectionString,
+                        containerName,
+                        new DataIdentifier(stripMetaKeyPrefix(blob.getName())),
+                        blob.getProperties().getLastModified().getTime(),
+                        blob.getProperties().getLength(),
+                        true));
+                }
+            }
+            LOG.debug("Metadata records read. recordsRead={} metadataFolder={} duration={}", records.size(), prefix, (System.currentTimeMillis() - start));
+        }
+        catch (StorageException e) {
+            LOG.info("Error reading all metadata records. metadataFolder={}", prefix, e);
+        }
+        catch (DataStoreException | URISyntaxException e) {
+            LOG.debug("Error reading all metadata records. metadataFolder={}", prefix, e);
+        }
+        finally {
+            if (null != contextClassLoader) {
+                Thread.currentThread().setContextClassLoader(contextClassLoader);
+            }
+        }
+        return records;
+    }
+
+    @Override
+    public boolean deleteMetadataRecord(String name) {
+        long start = System.currentTimeMillis();
+        ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
+        try {
+            Thread.currentThread().setContextClassLoader(getClass().getClassLoader());
+
+            CloudBlockBlob blob = getAzureContainer().getBlockBlobReference(addMetaKeyPrefix(name));
+            boolean result = blob.deleteIfExists();
+            LOG.debug("Metadata record deleted. existed={} metadataName={} duration={}", result, name, (System.currentTimeMillis() - start));
+            return result;
+
+        }
+        catch (StorageException e) {
+            LOG.info("Error deleting metadata record. metadataName={}", name, e);
+        }
+        catch (DataStoreException | URISyntaxException e) {
+            LOG.debug("Error deleting metadata record. metadataName={}", name, e);
+        }
+        finally {
+            if (contextClassLoader != null) {
+                Thread.currentThread().setContextClassLoader(contextClassLoader);
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public void deleteAllMetadataRecords(String prefix) {
+        if (null == prefix) {
+            throw new NullPointerException("prefix");
+        }
+        long start = System.currentTimeMillis();
+        ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
+        try {
+            Thread.currentThread().setContextClassLoader(getClass().getClassLoader());
+
+            CloudBlobDirectory metaDir = getAzureContainer().getDirectoryReference(META_DIR_NAME);
+            int total = 0;
+            for (ListBlobItem item : metaDir.listBlobs(prefix)) {
+                if (item instanceof CloudBlob) {
+                    ((CloudBlob)item).deleteIfExists();
+                    total ++;
```

Comment:
Since it might be possible to have total != listBlobs.length (since calls are async), the code was update to increase total only if delete succeeds.